### ISSUE TITLE
Bug: Connections to GitHub every three minutes?

### ISSUE THREAD

**Issue by mawmawmawm** [qdm12/private-internet-access-docker#166]
**TLDR**: *The docer containter connects to github every three minutes and I wonder why*

1. Is this urgent?

    - [x] No

2. What VPN service provider are you using?

    - [x] PIA

3. What's the version of the program?

    **See the line at the top of your logs**

`Running version latest built on 2020-05-18T13:39:35Z (commit 43c15b3)`

4. What are you using to run the container?

    - [x] Docker Compose

5. Extra information
Hi there, every since I started the container, the container seems to connect to github roughly every three minutes. The screenshot below is from Pi-Hole.
I wonder why this is the case and if there is anything here to fix this?
I verified this by stopping the container and the connections stopped.

![image](https://user-images.githubusercontent.com/9080951/83047752-aa3eb500-9ffd-11ea-8c5a-cd3dea1af422.png)

Host OS:
Synology NAS / DSM

**Comment by qdm12** [comment]
Oh I was myself wondering!

That's the docker healthcheck, see [this line](https://github.com/qdm12/private-internet-access-docker/blob/master/internal/healthcheck/healthcheck.go#L14) which resolves github.com and downloads github.com page over both http and https to make sure the connection is still up.

I think that's a good time to think about a better healthcheck, any suggestion? :wink:

**Comment by qdm12** [comment]
In my opinion, there are two things we want to check:

1. Your IP is still the vpn ip address
1. Your connection is still working

For 1. we can use several websites (ipinfo.io, ipify.com etc.) in a cycling fashion to avoid being blocked (too many attempts).

For 2. that would still need to access some internet server, like github.com. I picked github.com as it's accessed to update DNS block lists periodically so it would be the same more or less privacy wise.

I quite enjoy that healthcheck to know when my container is screwed up, but maybe should I add an option to disable it if you want slightly extra privacy?

Thanks

**Comment by Frepke** [comment]
you can setup a free account at https://ipdata.co/, they give you 1500 free requests a day.
Maybe you can make a variable, for the ones who want to use this, for their personal API.

**Comment by mawmawmawm** [comment]
Thanks all for clarifying this. Makes sense to have this health check but as mentioned above, it would be great if this would be configurable.

**Comment by qdm12** [comment]
Yes I'll make it configurable.

@Frepke thanks for the tip, although I'd rather cycle through no registration ip echo websites (for privacy and ease of setup). The cycling method is what I use for my [ddns updater](https://github.com/qdm12/ddns-updater) every 5 minutes and never got blocked :wink:

**Comment by Frepke** [comment]
> Yes I'll make it configurable.
> 
> @Frepke thanks for the tip, although I'd rather cycle through no registration ip echo websites (for privacy and ease of setup). The cycling method is what I use for my [ddns updater](https://github.com/qdm12/ddns-updater) every 5 minutes and never got blocked 😉

I was using ipinfo.io for checking my ip addresses from my vpn and linked containers. I also print the connected location. For me, with ipinfo.io, the location was always wrong. I took a free ipinfo account so I could ask them to check the wrong locations. Every time I asked them, they solved this issue within 24 hours. But after a dozen of times I quit contacting them and started using ipdata.co. They give me always the right information 😄 But for healthchecks, the location isn't that important.

**Comment by qdm12** [comment]
Actually @mawmawmawm  I just realised, how do you see these periodic DNS resolution on pihole? Are you not using the built-in DNS over TLS? If you do, then this is alarming I think!

**Comment by mawmawmawm** [comment]
I disabled DOH, that’s how I saw it

**Comment by qdm12** [comment]
@mawmawmawm you can disable the healthcheck with Docker actually, see [this](https://docs.docker.com/engine/reference/run/#healthcheck)

TL;DR: use `--no-healthcheck` or for docker-compose.yml:

```yml
    healthcheck:
      disable: true
```

**Comment by mawmawmawm** [comment]
Thanks!
